homeassistant/auth/providers/__init__.py

```diff
@@ -228,18 +229,29 @@ def __init__(self, auth_provider: AuthProvider) -> None:
                     reason='login_expired'
                 )
 
-            result = await auth_module.async_validation(
+            result = await auth_module.async_validate(
                 self.user.id, user_input)  # type: ignore
             if not result:
                 errors['base'] = 'invalid_code'
+                self.invalid_mfa_times += 1
```

Comment:
We shouldn't do this here. Instead, the login flow can just keep track how often it has received user input.

Reply:
This is for login flow, the mfa itself don't have login_flow instance, it just act as a validator